### ISSUE TITLE
PostfixAdmin 3.2 setup load error, file in setup.php misnamed.

### ISSUE THREAD

**Issue by AncientTom** [postfixadmin/postfixadmin#353]
The setup.php file on line 32 is trying to load '/../templates/header.php'. That sub-directory is now named templates_c. That line needs to be changed to '/../templates_c/header.php' and migrated to all the repositories.

**Comment by DavidGoodwin** [comment]
Err... templates_c should contain the 'compiled' smarty templates and shouldn't really be touched.

I do see https://github.com/postfixadmin/postfixadmin/blob/postfixadmin_3.2/templates/header.php - is there some packaging issue that's really the problem?

**Comment by AncientTom** [comment]
The package I have from the CentOS 8 repositories is all screwed up. Some php scripts are looking for templates and others are looking for templates_c. I tried to fix my issue by renaming the file but that caused other errors. The only work around I have is to add a link from one directory name to the other. Now it's working. (I hope) How does a bug like this find it's way out and into the repositories.

**Comment by DavidGoodwin** [comment]
(Copied from : 
https://sourceforge.net/p/postfixadmin/discussion/676076/thread/e6c33548/?limit=25 )

Symlinking templates_c to templates seems a really weird thing to do.

Smarty SHOULD compile things from 'templates' and store that compiled stuff in 'templates_c' (cached templates).

templates_c needs to be writeable by the webserver. Where it is has changed a little over the years - but you could have it set to /tmp/templates_c if you're not sharing the webserver with anything else.

Currently in version 3.2 we have this logic for templates_c -> which should be using a templates_c directory in the 'root' directory of your postfixadmin install. (e.g. /var/lib/postfixadmin/templates_c where /var/lib/postfixadmin/public is exposed via Apache).

https://github.com/postfixadmin/postfixadmin/blob/postfixadmin_3.2/lib/smarty.inc.php#L18


The templates_c directory needs to be writeable by the webserver, and you may find that something like SELinux blocks this (perhaps check /var/log/syslog or similar?). 

It's unlikely that Smarty would overwrite an existing .tpl file when it makes it's cached (templates_c) variant.  But it'll make a bit of a mess.

**Comment by ZsZs73** [comment]
Actually packagers are not up to date with the creation of templates_c directory. It is also missing from debian 10 package.
My workaround is:
```
mkdir -p /var/cache/postfixadmin/templates_c/
chown www-data /var/cache/postfixadmin/templates_c/
ln -s /var/cache/postfixadmin/templates_c /usr/share/postfixadmin/templates_c
```
Note that debian package puts postfixadmin related files under `/usr/share/postfixadmin`
I think that cache related files have better place under /var/cache

**Comment by AncientTom** [comment]
David, Thanks for your comment. This is all true but there are also php scripts that are trying to access the old /templates sub-directory. A deeper dive into PostfixAdmin needs to be done to clean things up and get all php devices on the same page before releasing and creating this headache for less technical users.

**Comment by AncientTom** [comment]
ZsZsZs, I also found a disagreement on the internet as to where to put PostfixAdmiin. By looking at several installation tutorials, There are 3 different directories that I found that installers are using. I went with the /usr/share/postfixAdmin home.

**Comment by DavidGoodwin** [comment]
Installation directory -> that's down to you or the distribution. 

You might have e.g. /srv/postfixadmin or /usr/share/postfixadmin  or something totally different. It doesn't matter too much, as long as your webserver is configured for it and can access the files.


I suspect there is "disagreement on the internet" because Postfixadmin has naturally changed over time, and some tutorials or guides on the internet may be out of date ... or may just be reflecting differences in packaging by various distributions.

I'm not convinced there is a need for a 'deeper dive' into PostfixAdmin to 'clean things up' - there should be only be one place where the templates and templates_c directories are defined by Postfixadmin itself (lib/inc.smarty.php).  But as always, if you can see a better way of doing things, feel free to submit a patch / pull request.

**Comment by AncientTom** [comment]
David; Just a case in point. the setup.php file is still looking for /templates. You first run setup.php and right off the bat, an error happens. Since php7 is shipped with display errors turned off, all you get is a blank page which is a stopper for most users. After looking for the problem, I changed the name. "/directory_c to "/directory" and setup then worked. after setting the admin password, I ran PostfixAdmin and got a blank page again. Rather than spending the rest of the day messing with it, I just used a symlink.

**Comment by DavidGoodwin** [comment]
What sort of messages did you see in the server's error log ? (E.g. /var/log/apache2/error.log ?)

**Comment by AncientTom** [comment]
David; Just a case in point. the setup.php file is still looking for /templates. You first run setup.php and right off the bat, an error happens. Since php7 is shipped with display errors turned off, all you get is a blank page which is a stopper for most users. After looking for the problem, I changed the name. "/directory_c to "/directory" and setup then worked. after setting the admin password, I ran PostfixAdmin and got a blank page again. Rather than spending the rest of the day messing with it, I just used a symlink.

**Comment by AncientTom** [comment]
I didn't look at /var/log/httpd/error_log because it was apparent that it was a php problem when I turned on display errors in php.ini and loaded the page again. This showed the php fatal error.

Anyway, I have moved on to debug my Postfix config problems since PostfixAdmin is now working for me. (I think) I created a bunch of vmail boxes with PostfixAdmin and it should have created those in /home/vmail/ which is empty. I am trying to find where PostfixAdmin put the domain mail directories.
